python/lsst/obs/test/ingestCalibs.py

```diff
@@ -0,0 +1,74 @@
+# This file is part of obs_test.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""Custom task for ingesting obs_test calibration files.
+"""
+
+__all__ = ["TestCalibsParseTask"]
+
+import os
+from lsst.pipe.tasks.ingestCalibs import CalibsParseTask
+
+
+class TestCalibsParseTask(CalibsParseTask):
+    """Specialization of `CalibsParseTask` for files with minimal headers.
+    """
+    def getCalibType(self, filename):
+        """Return a a known calibration dataset type using the filename.
+
+        Parameters
+        ----------
+        filename : `str`
+            Input filename
+
+        Returns
+        -------
+        type : one of "bias", "dark", "flat", "fringe", "sky", "defect", "unknown"
+            The type of dataset.
+        """
+        basename = os.path.basename(filename)
+        obstype = "unknown"
+        if "flat" in basename:
```

Comment:
Consider:

    _knownTypes = frozenset(("bias", "flat", "defect"))  # known file types as a class variable
    def getCalibType(...):
        ....
        typename = basename.split("_")[0]
        if typename in self._knownTypes:
            return typename
        return "unknown"

My main justification that this only examines the bit of the file name that actually contains the type. It's no a big deal since the names are so simple. I used a class variable so the set only had to be constructed once, but that is an optimization that is a bit silly for so few values. It would be plenty fast enough to use `if typename in ("bias", "flat", "defect"):` and not even bother with a set.

If you prefer the simplicity of examining the whole `basename`, consider eliminating repetition of those names with a loop:

    for typename in ("bias", "flat", "defect"):
        if typename in basename:
            return typename
    return "unknown"